### ISSUE TITLE
Alias domain support

### ISSUE THREAD

**Issue by pesimon** [Mailu/Mailu#200]
Some mail setups offer the possibility to create alias domains. It would be cool if mailu can implement this as well. A simple postfix implementation would look like [this](https://serverfault.com/a/813641) 

As far as I see, we would need to adjust the sqlite-virtual_alias_maps sql query, which is even a slightly complex sql ;-)
`virtual_alias_maps = ${sql}sqlite-virtual_alias_maps.cf`

```
  SELECT destination
   FROM
     (SELECT destination, email, wildcard, localpart FROM alias
      UNION
      SELECT email||(CASE WHEN forward_enabled=1 THEN ','||forward_destination ELSE '' END) AS destination, email, 0 as wildcard, localpart FROM user)
   WHERE
     (
      wildcard = 0
      AND
      email = '%s'
     ) OR (
      wildcard = 1
      AND
      '%s' LIKE email
     )
   ORDER BY
     wildcard ASC,
     length(localpart) DESC
   LIMIT 1
```
OR add another query plus maybe table for domain aliases since multiple sql maps [seam to be supported](https://workaround.org/ispmail/wheezy/connecting-postfix-to-the-database) 
 
What do you think? Or are there other ideas?

**Comment by curry684** [comment]
I actually need this feature to be able to transfer some of my older customers, so by all means prepare a PR for discussion   :)

**Comment by nalt** [comment]
I would also really like this feature. Did anyone test the proposed SQL?
I have been trying to get it working using virtual_alias_domains as an override, but this seems to interfere with the SQL commands.

**Comment by kaiyou** [comment]
@nalt, the proposed SQL does not solve the issue: we would first need to implement some admin objects and views for editing alias domains, then integrate them in SQL.

**Comment by kaiyou** [comment]
I just started implementing this in the `feature-alternative-domains` branch. I decided to name alias domain "alternatives" to avoid any confusion. Currently only the admin part is written and ready for testing, we will then have to update the Postfix configuration to reflect the changes.

**Comment by kaiyou** [comment]
The branch now has a full implementation of domain aliases. Because the virtual mailbox domain list is cached by Postfix, a new domain takes a couple seconds to be routed properly.

I am keeping this in a dedicated branch for some more testing before we merge in `master`.

**Comment by kaiyou** [comment]
This was merged as part of #252. Closing this issue.
